### PR TITLE
Bump alpine images versions

### LINKED ISSUE #152: bump alpine version
https://hub.docker.com/_/alpine

alpine 3.6, 3.9 passed.

Other failed:

alpine 3.12 failed with:
```
+ cp '.libs/libprofiler.so*' /usr/local/lib
cp: can't stat '.libs/libprofiler.so*': No such file or directory
The command '/bin/sh -c set -x     && apk add --no-cache --virtual .run-deps         libstdc++         readline         openssl         yaml         lz4         binutils         ncurses         libgomp         lua         tar         zip         zlib         libunwind         icu         ca-certificates     && apk add --no-cache --virtual .build-deps         gcc         g++         cmake         file         readline-dev         openssl-dev         yaml-dev         lz4-dev         zlib-dev         binutils-dev         ncurses-dev         lua-dev         musl-dev         make         git         libunwind-dev         autoconf         automake         libtool         linux-headers         go         icu-dev         wget     && : "---------- gperftools ----------"     && mkdir -p /usr/src/gperftools     && git clone "$GPERFTOOLS_REPO" /usr/src/gperftools     && git -C /usr/src/gperftools checkout "$GPERFTOOLS_TAG"     && (cd /usr/src/gperftools;         patch -p1 < /gperftools_alpine.diff;         rm /gperftools_alpine.diff;         ./autogen.sh;         ./configure;         make -j ;         cp .libs/libprofiler.so* /usr/local/lib;)     && (GOPATH=/usr/src/go go get github.com/google/pprof &&         cp /usr/src/go/bin/pprof /usr/local/bin || true )     && : "---------- tarantool ----------"     && mkdir -p /usr/src/tarantool     && git clone "$TARANTOOL_DOWNLOAD_URL" /usr/src/tarantool     && git -C /usr/src/tarantool checkout "$TARANTOOL_VERSION"     && git -C /usr/src/tarantool submodule update --init --recursive     && (cd /usr/src/tarantool;        echo "WARNING: Temporary fix for test/unit/cbus_hang test" ;        git cherry-pick d7fa6d34ab4e0956fe8a80966ba628e0e3f81067 2>/dev/null ||            git cherry-pick --abort ;        cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo             -DENABLE_BUNDLED_LIBYAML:BOOL=ON             -DENABLE_BACKTRACE:BOOL=ON             -DENABLE_DIST:BOOL=ON             .)     && make -C /usr/src/tarantool -j    && make -C /usr/src/tarantool install     && make -C /usr/src/tarantool clean     && : "---------- luarocks ----------"     && wget -O luarocks.tar.gz "$LUAROCKS_URL"     && mkdir -p /usr/src/luarocks     && tar -xzf luarocks.tar.gz -C /usr/src/luarocks --strip-components=1     && (cd /usr/src/luarocks;         ./configure;         make -j build;         make install)     && rm -r /usr/src/luarocks     && rm -rf /usr/src/tarantool     && rm -rf /usr/src/gperftools     && rm -rf /usr/src/go     && : "---------- remove build deps ----------"     && apk del .build-deps' returned a non-zero code: 1

```

alpine 3.11 failed with:
```
 ---> Running in e4cebacf4e64
+ apk add --no-cache --virtual .run-deps mariadb-client-libs libpq cyrus-sasl mosquitto-libs libev
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  mariadb-client-libs (missing):
    required by: .run-deps-20200607.121339[mariadb-client-libs]
The command '/bin/sh -c set -x     && apk add --no-cache --virtual .run-deps         mariadb-client-libs         libpq         cyrus-sasl         mosquitto-libs         libev     && apk add --no-cache --virtual .build-deps         git         cmake         make         coreutils         gcc         g++         postgresql-dev         lua-dev         musl-dev         cyrus-sasl-dev         mosquitto-dev         libev-dev         wget         unzip     && mkdir -p /rocks     && : "---------- proj (for gis module) ----------"     && wget -O proj.tar.gz http://download.osgeo.org/proj/proj-4.9.3.tar.gz     && mkdir -p /usr/src/proj     && tar -xzf proj.tar.gz -C /usr/src/proj --strip-components=1     && (cd /usr/src/proj;         ./configure;         make -j ;         make install)     && rm -r /usr/src/proj     && rm -rf /usr/src/proj     && rm -rf /proj.tar.gz     && : "---------- geos (for gis module) ----------"     && wget -O geos.tar.bz2 http://download.osgeo.org/geos/geos-3.6.0.tar.bz2     && mkdir -p /usr/src/geos     && tar -xjf geos.tar.bz2 -C /usr/src/geos --strip-components=1     && (cd /usr/src/geos;         ./configure;         make -j ;         make install)     && rm -r /usr/src/geos     && rm -rf /usr/src/geos     && rm -rf /geos.tar.bz2     && : "---------- luarocks ----------"     && cd /     && : "ldoc"     && tarantoolctl rocks install ldoc --server=http://rocks.moonscript.org     && : "lua-term"     && tarantoolctl rocks install lua-term     && : "avro"     && tarantoolctl rocks install avro-schema $LUAROCK_AVRO_SCHEMA_VERSION     && : "expirationd"     && tarantoolctl rocks install expirationd $LUAROCK_EXPERATIOND_VERSION     && : "queue"     && tarantoolctl rocks install queue $LUAROCK_QUEUE_VERSION     && : "connpool"     && tarantoolctl rocks install connpool $LUAROCK_CONNPOOL_VERSION     && : "vshard"     && tarantoolctl rocks install vshard $LUAROCK_VSHARD_VERSION     && : "http"     && tarantoolctl rocks install http $LUAROCK_HTTP_VERSION     && : "pg"     && tarantoolctl rocks install pg $LUAROCK_TARANTOOL_PG_VERSION     && : "mysql"     && tarantoolctl rocks install mysql $LUAROCK_TARANTOOL_MYSQL_VERSION     && : "memcached"     && tarantoolctl rocks install memcached $LUAROCK_MEMCACHED_VERSION     && : "prometheus"     && tarantoolctl rocks install prometheus $LUAROCK_TARANTOOL_PROMETHEUS_VERSION     && : "mqtt"     && tarantoolctl rocks install mqtt $LUAROCK_TARANTOOL_MQTT_VERSION     && : "gis"     && tarantoolctl rocks install gis $LUAROCK_TARANTOOL_GIS_VERSION     && : "gperftools"     && tarantoolctl rocks install gperftools $LUAROCK_TARANTOOL_GPERFTOOLS_VERSION     && : "---------- remove build deps ----------"     && apk del .build-deps' returned a non-zero code: 2
```

alpine 3.10 failed with:
```
+ apk add --no-cache --virtual .run-deps mariadb-client-libs libpq cyrus-sasl mosquitto-libs libev
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  mariadb-client-libs (missing):
    required by: .run-deps-20200607.123110[mariadb-client-libs]
The command '/bin/sh -c set -x     && apk add --no-cache --virtual .run-deps         mariadb-client-libs         libpq         cyrus-sasl         mosquitto-libs         libev     && apk add --no-cache --virtual .build-deps         git         cmake         make         coreutils         gcc         g++         postgresql-dev         lua-dev         musl-dev         cyrus-sasl-dev         mosquitto-dev         libev-dev         wget         unzip     && mkdir -p /rocks     && : "---------- proj (for gis module) ----------"     && wget -O proj.tar.gz http://download.osgeo.org/proj/proj-4.9.3.tar.gz     && mkdir -p /usr/src/proj     && tar -xzf proj.tar.gz -C /usr/src/proj --strip-components=1     && (cd /usr/src/proj;         ./configure;         make -j ;         make install)     && rm -r /usr/src/proj     && rm -rf /usr/src/proj     && rm -rf /proj.tar.gz     && : "---------- geos (for gis module) ----------"     && wget -O geos.tar.bz2 http://download.osgeo.org/geos/geos-3.6.0.tar.bz2     && mkdir -p /usr/src/geos     && tar -xjf geos.tar.bz2 -C /usr/src/geos --strip-components=1     && (cd /usr/src/geos;         ./configure;         make -j ;         make install)     && rm -r /usr/src/geos     && rm -rf /usr/src/geos     && rm -rf /geos.tar.bz2     && : "---------- luarocks ----------"     && cd /     && : "ldoc"     && tarantoolctl rocks install ldoc --server=http://rocks.moonscript.org     && : "lua-term"     && tarantoolctl rocks install lua-term     && : "avro"     && tarantoolctl rocks install avro-schema $LUAROCK_AVRO_SCHEMA_VERSION     && : "expirationd"     && tarantoolctl rocks install expirationd $LUAROCK_EXPERATIOND_VERSION     && : "queue"     && tarantoolctl rocks install queue $LUAROCK_QUEUE_VERSION     && : "connpool"     && tarantoolctl rocks install connpool $LUAROCK_CONNPOOL_VERSION     && : "vshard"     && tarantoolctl rocks install vshard $LUAROCK_VSHARD_VERSION     && : "http"     && tarantoolctl rocks install http $LUAROCK_HTTP_VERSION     && : "pg"     && tarantoolctl rocks install pg $LUAROCK_TARANTOOL_PG_VERSION     && : "mysql"     && tarantoolctl rocks install mysql $LUAROCK_TARANTOOL_MYSQL_VERSION     && : "memcached"     && tarantoolctl rocks install memcached $LUAROCK_MEMCACHED_VERSION     && : "prometheus"     && tarantoolctl rocks install prometheus $LUAROCK_TARANTOOL_PROMETHEUS_VERSION     && : "mqtt"     && tarantoolctl rocks install mqtt $LUAROCK_TARANTOOL_MQTT_VERSION     && : "gis"     && tarantoolctl rocks install gis $LUAROCK_TARANTOOL_GIS_VERSION     && : "gperftools"     && tarantoolctl rocks install gperftools $LUAROCK_TARANTOOL_GPERFTOOLS_VERSION     && : "---------- remove build deps ----------"     && apk del .build-deps' returned a non-zero code: 2
```

### PR DESCRIPTION
**Pull request by avtikhon**
Bumped alpine images versions.
Checked versions:
3.12 - failed
3.11 - failed
3.10 - failed
3.9 - passed
3.6 - passed
Decide to use 3.9. The next steps to fix the issue need to fix 3.12 version.
Also implemented ability to build images for different Docker's base images,
setting it in build options.
Tuned configuration to make all builds pass.

Depends on #161
Closes #152

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -34,7 +35,8 @@ before_script:
 'alpine 3.5 1.10.0':
   <<: *build_definition
   variables:
-    OS: 'alpine_3.5'
+    OS: 'alpine'
+    OS_VER: '3.5'
```
Comment (Totktonada) [inline .gitlab-ci.yml:39]: Nit: We usually use DIST name for this.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -14,12 +14,13 @@ before_script:
   stage: build
   tags:
     - deploy_test
+  interruptible: true
```
Comment (Totktonada) [inline .gitlab-ci.yml:17]: It looks unrelated here.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -144,18 +157,20 @@ before_script:
 'alpine 3.5 2.1':
   <<: *build_definition
   variables:
-    OS: 'alpine_3.5'
+    OS: 'alpine'
+    OS_VER: '3.9'
     TAG: '2.1'
     VER: '2.1'
-    DVER: '2.2'
+    DVER: '1.x'
```
Comment (Totktonada) [inline .gitlab-ci.yml:164]: What is DVER? Why it is changed here?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -191,7 +209,8 @@ before_script:
 'alpine 3.5 2.2':
```
Comment (Totktonada) [inline .gitlab-ci.yml:209]: The job name should be updated: 3.5 → 3.9.

File: README.md [hunk 5]
```diff
@@ -311,42 +311,40 @@ $ docker run -it tarantool/tarantool:2
...
-| 2.2        | dockerfile/alpine_3.5_2.x |
-| 2.3        | dockerfile/alpine_3.5_2.x |
-| 2.4        | dockerfile/alpine_3.5_2.x |
-| 2/latest   | dockerfile/alpine_3.5_2.x |
+| 1          | dockerfile/alpine_3.9_1.x |
+| 2.1        | dockerfile/alpine_3.9_1.x |
+| 2.2        | dockerfile/alpine_3.9_2.x |
+| 2.3        | dockerfile/alpine_3.9_2.x |
+| 2.4        | dockerfile/alpine_3.9_2.x |
+| 2/latest   | dockerfile/alpine_3.9_2.x |
```
Comment (Totktonada) [inline README.md:347]: Raw idea: Maybe update only rolling versions for now, but add future fixed ones based on Alpine 3.9/3.12? This way looks more safe for me.

File: dockerfiles/alpine_3.5_1.10.3 [hunk 6]
```diff
@@ -1,4 +1,5 @@
-FROM alpine:3.5
+ARG IMG
```
Comment (Totktonada) [inline dockerfiles/alpine_3.5_1.10.3:1]: Maybe name it more descriptive, like `BASE_IMAGE`?

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -39,7 +39,7 @@ before_script:
     OS_VER: '3.5'
     TAG: '1.10.0'
     VER: '1.10.0'
-    DVER: '1.10.3'
+    DVER: '1.x'
```
Comment (Totktonada) [inline .gitlab-ci.yml:42]: Is it a kind of docker file to use, which works like a template specialized by providing `--build-arg` options? If so, I understand the change and it looks okay. But the variable should have more meaningful name. Like, maybe, DOCKERFILE_NAME_SUFFIX.

File: .gitlab-ci.yml (hunk 1 above)
Comment (avtikhon) [reply .gitlab-ci.yml:40]: Ok, changed.

File: .gitlab-ci.yml (hunk 2 above)
Comment (avtikhon) [reply .gitlab-ci.yml:17]: This is the hidden job used as template for all jobs. The main idea is to skip building jobs in queue if the newer same jobs exists.

File: .gitlab-ci.yml (hunk 3 above)
Comment (avtikhon) [reply .gitlab-ci.yml:165]: DVER means dockerfile version, it shows the dockerfile to be used for building.

File: README.md (hunk 5 above)
Comment (avtikhon) [reply README.md:347]: Alpine 3.9 is the oldest release version mentioned as supported on alpine site, so later will update only rolling versions, but for now we need to update it ones.

File: .gitlab-ci.yml (hunk 7 above)
Comment (avtikhon) [reply .gitlab-ci.yml:43]: Right, ok, naming changed.

File: .gitlab-ci.yml (hunk 2 above)
Comment (Totktonada) [reply .gitlab-ci.yml:17]: I understand, but it does not look related here anyway. This PR is about apline version bump, right?

File: .gitlab-ci.yml (hunk 3 above)
Comment (Totktonada) [reply .gitlab-ci.yml]: The commit is named as 'Bump alpine images version'. Why DVER is changed here from 2.2 to 1.x?

File: README.md (hunk 5 above)
Comment (Totktonada) [reply README.md:347]: But some fixed versions are on Alpine 3.5 anyway, so 'we cannot offer images based on an unsupported Apline version' would be too pathetic.

Up to you, but I still think that bump it in the rolling and future fixed versions is more safe way.

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -37,80 +37,80 @@ before_script:
   <<: *build_definition
   variables:
     OS: 'alpine'
-    OS_VER: '3.5'
+    DIST: '3.5'
```
Comment (Totktonada) [inline .gitlab-ci.yml:40]: This variable was introduced right in this PR. There is no reason to give it OS_VER name and then rename to DIST. Let's introduce it as DIST from scratch. (I understand, rebasing is not fun, but see, the result will be more nice.)

The same for BASE_IMAGE.

File: .gitlab-ci.yml (hunk 3 above)
Comment (avtikhon) [reply .gitlab-ci.yml]: Moved to standalone PR.

Comment (Totktonada) [reply .gitlab-ci.yml]: But it still present here.

File: dockerfiles/alpine_3.9_1.x [hunk 9]
```diff
@@ -94,6 +95,7 @@ RUN set -x \
        export PATH=${GOPATH}/bin:/usr/local/go/bin:$PATH && \
        export GOBIN=$GOROOT/bin && \
        mkdir -p ${GOPATH}/src ${GOPATH}/bin && \
+       export GO111MODULE=on && \
```
Comment (Totktonada) [inline dockerfiles/alpine_3.9_1.x:98]: Unrelated change.

File: dockerfiles/alpine_3.9_2.x [hunk 10]
```diff
@@ -93,6 +94,7 @@ RUN set -x \
        export PATH=${GOPATH}/bin:/usr/local/go/bin:$PATH && \
        export GOBIN=$GOROOT/bin && \
        mkdir -p ${GOPATH}/src ${GOPATH}/bin && \
+       export GO111MODULE=on && \
```
Comment (Totktonada) [inline dockerfiles/alpine_3.9_2.x:97]: Same, unrelated change.

**Comment by avtikhon** [comment]
> Okay, except 2.2 → 1.x change in the first commit (which should land within PR #174) and two unrelated changes with GO111MODULE.

All corrected.
